zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/RoleAddEntityProcessor.java

```diff
@@ -111,7 +113,19 @@ final var record = command.getValue();
 
   @Override
   public void processDistributedCommand(final TypedRecord<RoleRecord> command) {
-    stateWriter.appendFollowUpEvent(command.getKey(), RoleIntent.ENTITY_ADDED, command.getValue());
+    final var record = command.getValue();
+    roleState
+        .getEntityType(record.getRoleKey(), record.getEntityKey())
+        .ifPresentOrElse(
+            entityType -> {
+              final var errorMessage =
+                  ENTITY_ALREADY_ASSIGNED_ERROR_MESSAGE.formatted(
+                      record.getEntityKey(), record.getRoleKey());
+              rejectionWriter.appendRejection(command, RejectionType.INVALID_STATE, errorMessage);
```

Comment:
🔧 Would be better to set rejection type as `ALREADY_EXISTS` to align with other distribution rejections. (e.g. in `TenantAddEntityProcessor`)